docs/configuration.md
# Configurations

- [Introduction](#intro)

{#intro}
## [](#intro) Introduction
All of the configuration files in the root directory. The default config file is
`config.yml` is always loaded. Environment specific config files (e.g.:
`config.dev.yml`) can override any option in the default file. Each option is
documented, so feel free to look through the files and get familiar with the
options available to you.

Here is the default config file included with sereno.

``` yml
#--------------------------------------------------------------------------
# Website Name
#--------------------------------------------------------------------------
#
# This value is the name of your website. This value is used to Created
# title for each page. It is also used in navigation bar. For personal
# website, it should be set to your full name, (e.g. Rahul Kadyan).
#
siteName: My Serene Website

#--------------------------------------------------------------------------
# Website Description
#--------------------------------------------------------------------------
#
# This value is the short description of your website. This value is
# used to provide meta information to search engines or websites.
# Try to keep it short (upto 200 characters) and simple.
#
siteDescription: My space on the internet - another Sereno website.

#--------------------------------------------------------------------------
# Github Pages
#--------------------------------------------------------------------------
#
# Use Github Pages to publish your website for free.
#
github:
    #----------------------------------------------------------------------
    # Repository
    #----------------------------------------------------------------------
    #
    # The `repository` where the project is hosted.
    #
    # repository: git@github.com:user/repository.git

    #----------------------------------------------------------------------
    # Branch
    #----------------------------------------------------------------------
    #
    # The `branch` value is used to publish your website on Github Pages.
    #
    # Possible values for `branch`:
    #   - master
    #     For your personal website.
    #     Name of the repository is `<your github username>.github.io`.
    #     Your default working branch should be 'dev'.
    #
    #   - gh-pages
    #     For project website.
    #     Your default working branch should be 'master'.
    #
    branch: gh-pages

#--------------------------------------------------------------------------
# Markdown Configuration for Website
#--------------------------------------------------------------------------
#
# The `extends` value sets the parent blade component for the markdown
# renderer. And `yields` value places the content in the designated
# place on the parent blade component.
#
view:
    extends: markdown
    yields: markdown-body

#--------------------------------------------------------------------------
# Blog
#--------------------------------------------------------------------------
#
# These settings are used to properly generate your  blog  using Sereno.
# If _blog directory is present in the `content` directory, then the
# blog is generated automatically.
#
blog:
    #----------------------------------------------------------------------
    # Directory
    #----------------------------------------------------------------------
    #
    # Here you may specify which directory (relative to content) to use
    # for generating your blog.
    #
    directory: blog

    #----------------------------------------------------------------------
    # URL prefix
    #----------------------------------------------------------------------
    #
    # Pages generated by Blog builder are prefixed with this value.
    #
    url_prefix: blog

    #----------------------------------------------------------------------
    # Articles Per Page
    #----------------------------------------------------------------------
    #
    # Blog builder creates an index of your articles, listed on the
    # landing page of your blog. This value controls number of articles
    # listed per page of the index.
    #
    postsPerPage: 15

    #----------------------------------------------------------------------
    # Markdown Configuration for Articles
    #----------------------------------------------------------------------
    #
    # Use these configurations for markdown pages in blog.
    #
    post:
        extends: blog.post
        yields: post-content

    #----------------------------------------------------------------------
    # Disqus Comments
    #----------------------------------------------------------------------
    #
    # You can add comments to your blog articles. Set `enable_comments` to
    # true and put your Disqus site ID in `disqus_id`.
    #
    enable_comments: true
    disqus_id: ''

#--------------------------------------------------------------------------
# Documentation
#--------------------------------------------------------------------------
#
# These settings are used to properly generate documentation using Sereno.
# If docs directory is present in the `content` directory, then the
# docs are generated automatically.
#
docs:
    #----------------------------------------------------------------------
    # Directory
    #----------------------------------------------------------------------
    #
    # Here you may specify which directory (relative to content) to use
    # for generating your docs.
    #
    directory: docs

    #----------------------------------------------------------------------
    # URL prefix
    #----------------------------------------------------------------------
    #
    # Pages generated by docs builder are prefixed with this value.
    #
    url_prefix: docs

    #----------------------------------------------------------------------
    # Index
    #----------------------------------------------------------------------
    #
    # This file is used to generate sidebar for docs page.
    #
    index: documentation

    #----------------------------------------------------------------------
    # Markdown Configuration for Documentation
    #----------------------------------------------------------------------
    #
    # Docs pages have a different style (inspired & stolen from Laravel).
    #
    extends: docs.base
    yields: doc_content

    #----------------------------------------------------------------------
    # Docs Landing Page
    #----------------------------------------------------------------------
    #
    # Here you may specify which file to load by default if user visits
    # docs landing page.
    #
    default: installation

sereno:
    #----------------------------------------------------------------------
    # Website URL
    #----------------------------------------------------------------------
    #
    # This URL is used by the Sereno to properly generate URLs when using
    # the building the website. You should set this to the root of your
    # website so that it is generates correct URLs for pages & assets.
    #
    url: /

    #----------------------------------------------------------------------
    # Extensions
    #----------------------------------------------------------------------
    #
    # Extensions add functionalities to Sereno. `DefaultExtension` provides
    # all the features offered by Sereno.
    #
    extensions:
        - Znck\Sereno\DefaultExtension::class,

    #----------------------------------------------------------------------
    # Website Content Directories
    #----------------------------------------------------------------------
    #
    # These directories would be searched recursively to generate your
    # website.
    #
    directory:
        - content

    #----------------------------------------------------------------------
    # Views
    #----------------------------------------------------------------------
    #
    # Override default views from themes.
    #
    views:
        - resources/views

    #----------------------------------------------------------------------
    # Special Files
    #----------------------------------------------------------------------
    #
    # Preserve names of these files.
    #
    special_files:
        - '404'

    #----------------------------------------------------------------------
    # Ignored Files
    #----------------------------------------------------------------------
    #
    # These files won't appear in your generated website.
    #
    ignore:
      - .DS_Store
```


<br><br><br>
- - - - - - - - - - - - -
[Edit this page on Github](https://github.com/znck/sereno.in/edit/master/docs/configurations.md)


content/_docs/configuration.md
# Configurations

- [Introduction](#intro)

{#intro}
## [](#intro) Introduction
All of the configuration files in the root directory. The default config file is
`config.php` is always loaded. Environment specific config files (e.g.:
`config.dev.php`) can override any option in the default file. Each option is
documented, so feel free to look through the files and get familiar with the
options available to you.


<br><br><br>
- - - - - - - - - - - - -
[Edit this page on Github](https://github.com/znck/sereno.in/edit/master/content/_docs/configurations.md)


config.php
<?php return [
    /*
    |--------------------------------------------------------------------------
    | Website Name
    |--------------------------------------------------------------------------
    |
    | This value is the name of your website. This value is used to Created
    | title for each page. It is also used in navigation bar. For personal
    | website, it should be set to your full name, (e.g. Rahul Kadyan).
    |
    */
    'siteName' => 'Sereno',

    /*
    |--------------------------------------------------------------------------
    | Website Description
    |--------------------------------------------------------------------------
    |
    | This value is the short description of your website. This value is
    | used to provide meta information to search engines or websites.
    | Try to keep it short (upto 200 characters) and simple.
    |
    */
    'siteDescription' => 'Keep thinking your story. We are gathering the world to listen to it.',

    /*
    |--------------------------------------------------------------------------
    | Github Pages
    |--------------------------------------------------------------------------
    |
    | The `branch` value is used to publish your website on Github Pages.
    |
    | Possible values for `branch`:
    |   - master
    |     For your personal website.
    |     Name of the repository is `<your github username>.github.io`.
    |     Your default working branch should be 'dev' (anything but not master).
    |
    |   - gh-pages
    |     For project website.
    |     Your default working branch should be 'master' (anything but not gh-pages).
    */
    'github' => [
        'branch' => 'gh-pages',
        'author' => 'Rahul Kadyan <hi@znck.me>',
    ],

    /*
    |--------------------------------------------------------------------------
    | Markdown Configuration for Website
    |--------------------------------------------------------------------------
    |
    | The `extends` value sets the parent blade component for the markdown
    | renderer. And `yields` value places the content in the designated
    | place on the parent blade component.
    |
    */
    'view' => [
        'extends' => 'markdown',
        'yields'  => 'markdown-body',
    ],


    /*
    |--------------------------------------------------------------------------
    | Blog
    |--------------------------------------------------------------------------
    |
    | These settings are used to properly generate your  blog  using Sereno.
    | If _blog directory is present in the `content` directory, then the
    | blog is generated automatically.
    |
    */
    'blog' => [
        /*
        |--------------------------------------------------------------------------
        | Directory
        |--------------------------------------------------------------------------
        |
        | Here you may specify which directory (relative to content) to use
        | for generating your blog.
        |
        */
        'directory' => 'blog',

        /*
        |--------------------------------------------------------------------------
        | URL prefix
        |--------------------------------------------------------------------------
        |
        | Pages generated by Blog builder are prefixed with this value.
        |
        */
        'url_prefix' => 'blog',
        /*
        |--------------------------------------------------------------------------
        | Articles Per Page
        |--------------------------------------------------------------------------
        |
        | Blog builder creates an index of your articles, listed on the landing
        | page of your blog. This value controls number of articles listed
        | per page of the index.
        |
        */
        'postsPerPage' => 15,

        /*
        |--------------------------------------------------------------------------
        | Markdown Configuration for Articles
        |--------------------------------------------------------------------------
        |
        | Use these configurations for markdown pages in blog.
        |
        */
        'post' => [
            'extends' => 'blog.post',
            'yields'  => 'post-content',
        ],

        /*
        |--------------------------------------------------------------------------
        | Disqus Comments
        |--------------------------------------------------------------------------
        |
        | You can add comments to your blog articles. Set `enable_comments` to
        | true and put your Disqus site ID in `disqus_id`.
        |
        */
    	'enable_comments' => true,
        'disqus_id' => 'sereno-1',
    ],

    /*
    |--------------------------------------------------------------------------
    | Documentation
    |--------------------------------------------------------------------------
    |
    | These settings are used to properly generate documentation using Sereno.
    | If docs directory is present in the `content` directory, then the
    | docs are generated automatically.
    |
    */
    'docs' => [
        /*
        |--------------------------------------------------------------------------
        | Directory
        |--------------------------------------------------------------------------
        |
        | Here you may specify which directory (relative to content) to use
        | for generating your docs.
        |
        */
        'directory'  => 'docs',

        /*
        |--------------------------------------------------------------------------
        | URL prefix
        |--------------------------------------------------------------------------
        |
        | Pages generated by docs builder are prefixed with this value.
        |
        */
        'url_prefix' => 'docs',

        /*
        |--------------------------------------------------------------------------
        | Index
        |--------------------------------------------------------------------------
        |
        | This file is used to generate sidebar for docs page.
        |
        */
        'index' => 'documentation',

        /*
        |--------------------------------------------------------------------------
        | Markdown Configuration for Documentation
        |--------------------------------------------------------------------------
        |
        | Docs pages have a different style (inspired & stolen from Laravel).
        |
        */
        'extends' => 'docs.base',
        'yields'  => 'doc_content',

        /*
        |--------------------------------------------------------------------------
        | Docs Landing Page
        |--------------------------------------------------------------------------
        |
        | Here you may specify which file to load by default if user visits
        | docs landing page.
        |
        */
        'default' => 'installation',
    ],

    'sereno' => [

        /*
        |--------------------------------------------------------------------------
        | Website URL
        |--------------------------------------------------------------------------
        |
        | This URL is used by the Sereno to properly generate URLs when using
        | the building the website. You should set this to the root of your
        | website so that it is generates correct URLs for pages & assets.
        |
        */
        'url' => 'http://sereno.in/',

        /*
        |--------------------------------------------------------------------------
        | Commands
        |--------------------------------------------------------------------------
        |
        | Add commands to Sereno CLI.
        |
        */
        'commands' => [],

        /*
        |--------------------------------------------------------------------------
        | Extensions
        |--------------------------------------------------------------------------
        |
        | Extensions add functionalities to Sereno. `DefaultExtension` provides
        | all the features offered by Sereno.
        |
        */
        'extensions' => [
            Znck\Sereno\DefaultExtension::class,
        ],

        /*
        |--------------------------------------------------------------------------
        | Website Content Directories
        |--------------------------------------------------------------------------
        |
        | These directories would be searched recursively to generate your website.
        |
        */
        'directory' => [
            'content',
        ],

        /*
        |--------------------------------------------------------------------------
        | Views
        |--------------------------------------------------------------------------
        |
        | Override default views from themes.
        |
        */
        'views' => [
            'resources/views',
        ],

        /*
        |--------------------------------------------------------------------------
        | Special Files
        |--------------------------------------------------------------------------
        |
        | Preserve names of these files.
        |
        */
        'special_files' => [
            '404',
        ],

        /*
        |--------------------------------------------------------------------------
        | Ignored Files
        |--------------------------------------------------------------------------
        |
        | These files won't appear in your generated website.
        |
        */
        'ignore' => [],
    ],
];
